lib/language_pack/ruby.rb

```diff
@@ -412,7 +412,7 @@ def setup_profiled(ruby_layer_path: , gem_layer_path: )
       set_env_override "DISABLE_SPRING", "1"
 
       set_env_default "MALLOC_ARENA_MAX", "2"     if default_malloc_arena_max?
-      add_to_profiled set_default_web_concurrency if env("SENSIBLE_DEFAULTS")
+      add_to_profiled (env("SENSIBLE_DEFAULTS") ? set_default_web_concurrency : ""), "WEB_CONCURRENCY.sh", "w" # always write that file, even if its empty (meaning no defaults apply), for interop with other buildpacks - and we overwrite that file rather than appending (which is the default)
```

Comment:
With the additional arguments and the ternary it makes this quite the line. Since `env("SENSIBLE_DEFAULTS")` should return nil we could short-circuit it with `env("SENSIBLE_DEFAULTS") || ""` which is a little more idiomatic. I'm not sure how I would write that up in a convention though.

Reply:
but we want `set_default_web_concurrency` if `SENSIBLE_DEFAULTS` is set, no?

I just realized this (new and old) check also doesn't cover `SENSIBLE_DEFAULTS=0`...

Reply:
> but we want set_default_web_concurrency if SENSIBLE_DEFAULTS is set, no?

Ahh, right. Sorry got my wires crossed. I want to get rid of the ternary, it's already confused me in my own review 🤪. We've got some options then. 

We could put the logic to decide to return the script or an empty string inside of the poorly named `set_default_web_concurrency` method.

```ruby
def set_default_web_concurrency # probably worth renaming to something like "default_web_concurrency"
  return "" if !env("SENSIBLE_DEFAULTS")
  # ...

end
```

> I just realized this (new and old) check also doesn't cover SENSIBLE_DEFAULTS=0...

That's logically consistent with Ruby. Any value that's not empty is truthy. It's common practice to just check for env var presence (for boolean env vars) rather than the specific contents "disabled,false,0,wrong,off,bad,etc." for example https://github.com/codetriage/CodeTriage/blob/main/config/environments/production.rb#L27. It does mean we have to a bit more cautious with our interfaces like this one where a customer essentially can never opt out of a default value unless they do something like this in their Procfile:

```
web: unset WEB_CONCURRENCY && ...
```

This is one of the prime hesitations setting a default for ruby and why these actions were originally guarded behind the (also poorly named SENSIBLE_DEFAULTS env var).

Reply:
Yeah I know that about Ruby, it's just that most programs that read env vars like this will usually treat an unset variable the same as one set to "0"...

(that example, by the way, uses Rails' `present?`, which returns `false` for an empty or blank whitespace string, which is distinct from our current behavior, which evaluates to `true` in that case... `FOO= ruby -e 'puts !!ENV["FOO"]'`)

Anyway, it definitely doesn't belong into `set_default_web_concurrency`, that function should only be concerned with generating the shell code, and the calling code decides when to call it.

Reply:
alright check out the updated version... ternary is still in but moved to its own line

Reply:
> it definitely doesn't belong into set_default_web_concurrency

This whole thing is very poorly architected in that it wasn't really architected at all. It's not really any less consistent to put the check in there than not. Moving to it's own line is good though. Let's ship it.